python/ModelTools.py

```diff
@@ -278,12 +259,37 @@ def doNuisances(self):
             self.doObj("nuisancePdf", "PROD", ",".join(["%s_Pdf" % n for (n,nf,p,a,e) in self.DC.systs]))
             self.doSet("globalObservables", ",".join(globalobs))
 	
+    def doNuisancesGroups(self):
+        # Prepare a dictionary of which group a certain nuisance belongs to
+        groupsFor = {}
+        existingNuisanceNames = tuple(set([syst[0] for syst in self.DC.systs]+self.DC.flatParamNuisances.keys()))
```

Comment:
Hi,

I can surely do that.

Probably in a different loop since discrete nuisances are RooCategories and
cannot be retrieve with ws.var.

Also the code that does the freezing should RooCategories cannot be set
const AFIUK so the freezing code should be aware of this. Is it the case?

Cheers,
Pasquale


On 9 July 2015 at 16:36, André David <notifications@github.com> wrote:

> In python/ModelTools.py
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/225#discussion_r34261524>
> :
>
> > @@ -278,12 +259,37 @@ def doNuisances(self):
> >              self.doObj("nuisancePdf", "PROD", ",".join(["%s_Pdf" % n for (n,nf,p,a,e) in self.DC.systs]))
> >              self.doSet("globalObservables", ",".join(globalobs))
> >  	
> > +    def doNuisancesGroups(self):
> > +        # Prepare a dictionary of which group a certain nuisance belongs to
> > +        groupsFor = {}
> > +        existingNuisanceNames = tuple(set([syst[0] for syst in self.DC.systs]+self.DC.flatParamNuisances.keys()))
>
> @nucleosynthesis <https://github.com/nucleosynthesis> proposed to also
> add self.DC.discretes to this expanded list.
> discretes is filled in
> https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/blob/slc6-root5.34.17/python/DatacardParser.py#L180-L183
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/225/files#r34261524>
> .
>



-- 
Pasquale Musella - ETHZ
http://musella.web.cern.ch/musella